File: Data Structures & Algorithms/valid-binary-search-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        if not root: return True
        stack = [(root, float("-inf"), float("inf"))]
        while stack:
            node, min_val, max_val = stack.pop()
            if not (min_val < node.val < max_val):
                return False

            if node.left:
                stack.append((node.left, min_val, node.val))
            if node.right:
                stack.append((node.right, node.val, max_val))
        return True

File: Data Structures & Algorithms/reverse-string/submission-2.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        stack<char> stk;
        for (const auto c: s) {
            stk.push(c);
        }

        int i = 0;
        while (!stk.empty()) {
            s[i++] = stk.top();
            stk.pop();
        } 
    }
};